File: DSA_CSHARP/Recursion/Power.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DSA_CSHARP.Recursion;
internal class Power
{
    public int pow(int m, int n)
    {
        if (n > 0)
            return pow(m, n - 1) * m;
        return 1;
    }
    public string Main()
    {
        Console.WriteLine($"------------------------------------");
        Console.Write($"Enter 'M': ");
        var m = Convert.ToInt32(Console.ReadLine());

        Console.Write($"Enter 'N': ");
        var n = Convert.ToInt32(Console.ReadLine());

        var res = pow(m, n);

        Console.WriteLine($"{m} power {n} is: {res}");
        return res.ToString();

    }
}

File: DSA_PYTHON/Recursion/Power.py
class Power:
    def pow(self, m, n):
        if n > 0:
            return self.pow(m, n - 1) * m
        return 1

    def main(self):
        print("------------------------------------")
        print("Enter 'M'")
        m = int(input())

        print("Enter 'N'")
        n = int(input())

        res = self.pow(m, n)
        print(f"{m} power {n} is: {res}")
        return res